rust/agama-server/src/questions/web.rs

```diff
@@ -0,0 +1,271 @@
+            dbus_password
+                .set_password(password.password.as_str())
+                .await?
+        }
+        let dbus_generic = GenericQuestionProxy::builder(&self.connection)
+            .path(&question_path)?
+            .cache_properties(zbus::CacheProperties::No)
+            .build()
+            .await?;
+        dbus_generic
+            .set_answer(answer.generic.answer.as_str())
+            .await?;
+        Ok(())
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum QuestionsError {
+    #[error("Question service error: {0}")]
+    Error(#[from] ServiceError),
+}
+
+impl IntoResponse for QuestionsError {
+    fn into_response(self) -> Response {
+        let body = json!({
+            "error": self.to_string()
+        });
+        (StatusCode::BAD_REQUEST, Json(body)).into_response()
+    }
+}
+
+#[derive(Clone)]
+struct QuestionsState<'a> {
+    questions: QuestionsClient<'a>,
+}
+
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct Question {
+    generic: GenericQuestion,
+    with_password: Option<QuestionWithPassword>,
+}
+
+/// Facade of agama_lib::questions::GenericQuestion
+/// For fields details see it.
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct GenericQuestion {
+    id: u32,
+    class: String,
+    text: String,
+    options: Vec<String>,
+    default_option: String,
+    data: HashMap<String, String>,
+}
+
+/// Facade of agama_lib::questions::WithPassword
+/// For fields details see it.
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct QuestionWithPassword {
+    password: String,
+}
+
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct Answer {
+    generic: GenericAnswer,
+    with_password: Option<PasswordAnswer>,
+}
+
+/// Answer needed for GenericQuestion
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct GenericAnswer {
+    answer: String,
+}
+
+/// Answer needed for Password specific questions.
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct PasswordAnswer {
+    password: String,
+}
+/// Sets up and returns the axum service for the questions module.
+pub async fn questions_service(dbus: zbus::Connection) -> Result<Router, ServiceError> {
+    let questions = QuestionsClient::new(dbus.clone()).await?;
+    let state = QuestionsState { questions };
+    let router = Router::new()
+        .route("/", get(list_questions))
+        .route("/:id/answer", put(answer))
+        .with_state(state);
+    Ok(router)
+}
+
+pub async fn questions_stream(
+    dbus: zbus::Connection,
+) -> Result<Pin<Box<dyn Stream<Item = Event> + Send>>, Error> {
+    let question_path = OwnedObjectPath::from(
+        ObjectPath::try_from("/org/opensuse/Agama1/Questions")
+            .context("failed to create object path")?,
+    );
+    let proxy = ObjectManagerProxy::builder(&dbus)
+        .path(question_path)
+        .context("Failed to create object manager path")?
```

Comment:
To be honest, I am still not a big fan of using `anyhow` in libraries. But I can leave with that :-)

Reply:
me neither, but it is much faster then adding each zvariant error :)